test/tests/parallel/set2/template.bats

```diff
@@ -0,0 +1,55 @@
+load "${REPOSITORY_ROOT}/test/test_helper/common"
+
+# ? global variable initialization
+# ?   currenlty here for transition from $NAME
+IMAGE_NAME=${NAME:-mailserver-testing:ci}
+# ?   to identify the test easily
+TEST_NAME_PREFIX='checking template: '
+# ?   must be unique
+CONTAINER_NAME='dms-test-template'
+# ?   use this to execute someting inside the container
+RUN_COMMAND=('run' 'docker' 'exec' "${CONTAINER_NAME}")
+
+# ? test setup
+
+function setup_file() {
+  # ? optional setup before container is started
+
+  # ? start of the container
+  # ?   PRIVATE_CONFIG must be set properly
+  local PRIVATE_CONFIG
+  PRIVATE_CONFIG=$(duplicate_config_for_container . "${CONTAINER_NAME}")
+
+  # ?   the run command should look like this, and only the `--env`
+  # ?   should differe from test file to test file; if not needed,
+  # ?   disable AV / anti-spam software
+  docker run --rm --detach --tty \
+    --name "${CONTAINER_NAME}" \
+    --hostname mail.my-domain.com \
+    --volume "${PRIVATE_CONFIG}:/tmp/docker-mailserver" \
+    --env ENABLE_AMAVIS=0 \
+    --env ENABLE_CLAMAV=0 \
+    --env ENABLE_UPDATE_CHECK=0 \
+    --env ENABLE_SPAMASSASSIN=0 \
+    --env ENABLE_FAIL2BAN=0 \
+    --env LOG_LEVEL=debug \
```

Comment:
I noticed you added most of these ENV to some other tests you've updated (_two other files I think_).

Most of those are default `0`, but if you'd like to encourage a common base of ENV here, using the helper I had setup a few tests to use is a good way to do that.

They could probably be moved into that helpers base ENV, and this test adapted to use the helper. That does not need to happen for this PR though, which already has a lot going on.

I do like that it's more clear here, but I'm less fond of the copy/paste that I assume would occur from this template being applied to a wide number of test files.

I have also been taking interest with the "new" `docker compose` v2 command and leveraging `compose.yml` with the features it offers (_it may have been called profiles, or overrides_), which allows for establishing a base config, and providing other configs that can override or extend that.

It may be nicer to manage (we can still create some ENV var like `CONTAINER_NAME` and `PRIVATE_CONFIG` for it to use AFAIK, but avoid copy/paste, and the awkward helper approach I initially implemented.

If I have time in dec/jan I may take a shot at it, but anyone is welcome to investigate that if they've got the time :muscle:

Reply:
You're right, doing this in a common helper or with Compose v2 is better. I will revert these changes for default-values.